### DISCUSSION TITLE
Drawing stripped and dodged barplot

### DISCUSSION THREAD

**Discussion by harrysw1729** [JuliaPlots/Makie.jl#3868] [Q&A]
This is just a repeat of my unanswered question at [Discourse](https://discourse.julialang.org/t/drawing-stripped-dodged-bar-charts-in-makie/114208?u=harrysw). :(

In short, I tried to create a dodged barplot where the dodged bar would have different hatch patterns. Below is a short example of my attempt:

```julia
using CairoMakie

cats = repeat(Float64[1, 2], inner=2)
heights = repeat(Float64[1, 2], outer=2)
group = repeat([1, 2], outer=2)

directions = [[1.0, 1.0], [-1.0, 1.0]]
patterns = Any[Makie.LinePattern(background_color=(:red, 0.5), direction=direction) for direction in directions]

fig = Figure()
barplot(fig[1, 1], cats, heights, dodge=group, color=patterns[group])
fig
```

The code was based on the documentation about [dodged bars](https://docs.makie.org/stable/reference/plots/boxplot/) and [stripped bars](https://beautiful.makie.org/examples/2d/bars/stripped_bars). However, the code ran into the following error:

```julia
ERROR: StackOverflowError:
Stacktrace:
 [1] float32type(::Type{DataType}) (repeats 2 times)
   @ Makie ~/.julia/packages/Makie/eziUH/src/conversions.jl:706
```

The individual line patterns defined in my codes seemed to be fine in that the code would run without any issue I fix the line pattern, such as the one below.

```julia
fig2 = Figure()
barplot(fig2[1, 1], cats, heights, dodge=group, color=patterns[1])
fig2
```

Below is the version information:
```julia
versioninfo()

Julia Version 1.10.3
Commit 0b4590a5507 (2024-04-30 10:59 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: macOS (arm64-apple-darwin22.4.0)
  CPU: 8 × Apple M1
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, apple-m1)
Threads: 1 default, 0 interactive, 1 GC (on 4 virtual cores)
Environment:
  JULIA_EDITOR = code
  JULIA_NUM_THREADS =
```

And Makie info:

```julia
using Pkg
       Pkg.status(["CairoMakie", "Makie"])

Status `~/.julia/environments/v1.10/Project.toml`
  [13f3f980] CairoMakie v0.12.0
  [ee78f7c6] Makie v0.21.0
```

**Comment by asinghvi17** [comment]
Could you post the full error, including the stacktrace, along with which versions of Makie and Julia you're running?

You can get the versions by:
```julia
versioninfo()
```
and
```julia
using Pkg
Pkg.status(["CairoMakie", "Makie"])
```

**Reply by harrysw1729** [reply]
I just edited the post to include the information. Thanks a lot!

I also realized that my CairoMakie was one version behind. Strangely, after I updated it, the error message changed. 😅

**Comment by ffreyer** [comment]
I think you're running into two issues:
1. Patterns are currently one-per-plot
2. The `Any[]` might result in Makie going down the wrong conversion path as that's type based.

**Reply by harrysw1729** [reply]
Thanks for the reply. I tried removing `Any` but the StackTrace and the error message remained the same.

As for your point 1, could you please elaborate on that? Any suggestions on fixing the issue?
